RealmSwift/Sync.swift

```diff
@@ -277,17 +304,23 @@ public enum ClientResetMode {
 
     /**
      An enum which determines file recovery behvaior in the event of a client reset.
-     - note: Defaults to `.manual`
+     - note: Defaults to `.recover
 
      - see: `ClientResetMode` and `RLMClientResetMode`
      - see: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
     */
     public var clientResetMode: ClientResetMode {
         switch config.clientResetMode {
         case .manual:
-            return .manual
+            return .manual(config.manualClientResetHandler)
         case .discardLocal:
```

Comment:
I would probably change this to
```
case .discardLocal, .discardUnsyncedChanges:
            return .discardUnsyncedChanges(ObjectiveCSupport.convert(object: config.beforeClientReset), ObjectiveCSupport.convert(object: config.afterClientReset))
```

Reply:
RLMClientResetDiscardLocal and RLMClientResetDiscardUnsyncedChanges both assert to the same core enum. So the switch cases get obscured. Is someone is using .discardLocal and have switch cases for discardLocal, then those cases won't hit. That'd be a breaking change for them.

But that means for anyone using discardUnsyncedChanges, their cases hit on discardLocal but not discardUnsyncedChanges (see testClientResetDiscardUnsyncedChanges).

How can we differentiate? I tried using raw values in the swift enum, but you can't have a raw value _and_ enum parameters.

Reply:
Adding cases to an enum is already a breaking change because any code which switches on the enum will no longer work. As a result, returning a different enum case than we used to isn't really "more breaking".